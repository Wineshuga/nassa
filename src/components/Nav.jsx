import { NavLink } from 'react-router-dom';
import '../styles/nav.css';
import 'bootstrap/dist/css/bootstrap.css';

const links = [
  { path: '/', text: 'Rockets' },
  { path: '/Mission', text: 'Mission' },
  { path: '/MyProfile', text: 'My Profile' },
];
const Nav = () => (
  <>
    <section className="nav-container">
      <div>
        <img
          className="Logo"
          src="https://spacetravellerhubcapstone.netlify.app/static/media/planet.3f6fb2742f28651730c0.png"
          alt="Logo"
          style={{ width: '50px' }}
        />
        <h3>Space Travelers&apos; Hub</h3>
      </div>
      <div>
        <ul
          className="list-unstyled m-2 p-2 d-flex flex-wrap"
        >
          {links.map((link) => (
            <li key={link.text} className="m-2">
              <NavLink
                className={({ isActive }) => (isActive ? 'active-link' : 'link')}
                to={link.path}
              >
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>

      </div>
    </section>
    <hr className="m-3 my-1 py-2" />
  </>
);

export default Nav;
