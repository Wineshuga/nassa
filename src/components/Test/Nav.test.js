import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Nav from '../Nav';

test('Nav component matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Nav />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
