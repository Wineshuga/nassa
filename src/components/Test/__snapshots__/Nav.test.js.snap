// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-wrap align-items-center p-3 container"
  >
    <img
      alt="Logo"
      class="Logo"
      src="https://spacetravellerhubcapstone.netlify.app/static/media/planet.3f6fb2742f28651730c0.png"
      style="width: 50px;"
    />
    <h3>
      Space Travelers' Hub
    </h3>
    <ul
      class="list-unstyled m-2 p-2 d-flex flex-wrap"
      style="justify-content: flex-end; flex: 2;"
    >
      <li
        class="m-2"
      >
        <a
          aria-current="page"
          class="active-link"
          href="/"
        >
          Rockets
        </a>
      </li>
      <li
        class="m-2"
      >
        <a
          class="link"
          href="/Mission"
        >
          Mission
        </a>
      </li>
      <li
        class="m-2"
      >
        <a
          class="link"
          href="/MyProfile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
